### ISSUE TITLE
Improve documentation to make classes use the "public" module

### ISSUE THREAD

**Issue by CaselIT** [sqlalchemy/sqlalchemy#8894]
Discussed in #8893 

Currently it seems that sphinx uses a different path than the one of the definition, but it's still not the most common one for some classes.

For example `Column` is documented as `sqlalchemy.schema.Column`, while the most common one is `sqlalchemy.Column`.
In all cases the internal `sqlalchemy.sql.schema.Column` should not be used in the documentation or users

**Comment by zzzeek** [comment]
yeah I think we have to go into the .rst files and just change everything that's currently sqlalchemy.types, sqlalchemy.schema, sqlalchemy.engine, sqlalchemy.pool, sqlalchemy.event, sqlalchemy.sql.expression to just be "sqlalchemy".   I've never really wanted to do this because it always seemed good that the docs would show you where these things really are, but in 2.0 and especially with typing, I just want all those symbols and types to be under "sqlalchemy.".

**Comment by CaselIT** [comment]
can't we just change this https://github.com/sqlalchemy/sqlalchemy/blob/main/doc/build/conf.py#L136?

**Comment by CaselIT** [comment]
or maybe this https://github.com/sqlalchemy/sqlalchemy/blob/main/doc/build/conf.py#L171

**Comment by zzzeek** [comment]
those have to be changed sure, but we might also have to change it directly like here: https://github.com/sqlalchemy/sqlalchemy/blame/main/doc/build/core/metadata.rst#L11

**Comment by LeonxLJX** [comment]
I'd like to take this one.

Reading the issue, this looks like a documentation-clarity gap rather than a code defect: the current page doesn't give readers enough to self-diagnose, which is why it surfaces here as an issue instead of being answered by the docs.

Plan: locate the relevant section, confirm what the code actually does today (so the docs match behavior, not intent), tighten the wording, and scan neighboring pages for the same inconsistency — otherwise the fix just relocates the confusion.

One thing to confirm before writing: minimal correction scoped to this page, or also cross-link from related guides if the same gap shows up there?

(claiming via @LeonxLJX)
